rasa/core/tracker_store.py

```diff
@@ -162,17 +163,17 @@ def create(
         import sqlalchemy.exc
 
         try:
-            tracker_store = _create_from_endpoint_config(obj, domain, event_broker)
-            if not check_if_tracker_store_async(tracker_store):
-                rasa.shared.utils.io.raise_deprecation_warning(
-                    f"Tracker store implementation "
-                    f"{tracker_store.__class__.__name__} "
-                    f"is not asynchronous. Non-asynchronous tracker stores "
-                    f"are currently deprecated and will be removed in 4.0. "
-                    f"Please make the following methods async: "
-                    f"{_get_async_tracker_store_methods()}"
-                )
-                tracker_store = AwaitableTrackerStore(tracker_store)
+            auth_retry_wrapper = plugin_manager().hook.get_auth_retry_wrapper(
```

Comment:
style: this section reads a bit odd: a trackerstore is either an `auth_retry_wrapper` or a `create_tracker_store` 🤔 

maybe the hook should be called `create_tracker_store`?